### ISSUE TITLE
Glossary of technical terms

### ISSUE THREAD

**Issue by chenglou** [reasonml/reasonml.github.io#369]
Thinking of having a section on it. Explained in layman terms. Topologically sorted so that subsequent terms are always only explained using preceeding ones (aka a single top-down reading needed).

**Comment by kyldvs** [comment]
We got a lot covered in https://reasonml.github.io/docs/en/overview and tried to keep it topo sorted, feel free to reopen if you meant something else or there are specific topics missing
